core/service/src/lib.rs

```diff
@@ -502,6 +495,27 @@ impl<Components: components::Components> Service<Components> {
 		})
 	}
 
+	/// Returns a reference to the config passed at initialization.
+	pub fn config(&self) -> &FactoryFullConfiguration<Components::Factory> {
+		&self.config
+	}
+
+	/// Returns a reference to the config passed at initialization.
+	///
+	/// > **Note**: This method is currently necessary because we extract some elements from the
+	/// >			configuration at the end of the service initialization. It is intended to be
+	/// >			removed.
+	pub fn config_mut(&mut self) -> &mut FactoryFullConfiguration<Components::Factory> {
```

Comment:
Should this be `#[deprecated]`?

Reply:
There's no way to replace it at the moment.

The configuration at the moment has a field that acts as some sort of "generic storage for variables you need to pass between the multiple stages of building a service", which is an extremely crappy way of dealing with the fact that we're having this weird `Factory` trait.